stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4860,80 +4860,62 @@ void SortOp::build(OpBuilder& builder, OperationState& state,
   state.addRegion();
 }
 
-LogicalResult SortOp::verify() {
-  Operation::operand_range operands = this->operands();
-  if (operands.empty()) return emitOpError("requires at least one input");
-
-  // TODO(antiagainst): verify partionally dynamic shapes
-  if (llvm::all_of(operands, [](Value operand) {
-        return operand.getType().cast<ShapedType>().hasRank();
-      })) {
-    ArrayRef<int64_t> inputShape =
-        (*operands.begin()).getType().cast<ShapedType>().getShape();
-
-    if (llvm::any_of(llvm::drop_begin(operands, 1), [&](Value operand) {
-          return operand.getType().cast<ShapedType>().getShape() != inputShape;
-        }))
-      return emitOpError("requires all inputs to have the same dimensions");
-
-    int64_t rank = inputShape.size();
-    int64_t cmpDim = dimension();
-    if (cmpDim < -rank || cmpDim >= rank)
-      return emitOpError("dimension attribute value must be in range [-")
-             << rank << ", " << rank << "), but found " << cmpDim;
-  }
-
-  Block& block = comparator().front();
-  size_t numOperands = getOperation()->getNumOperands();
-  if (block.getNumArguments() != 2 * numOperands)
-    return emitOpError("comparator block should have ")
-           << 2 * numOperands << " arguments";
+LogicalResult SortOp::inferReturnTypeComponents(
+    MLIRContext*, Optional<Location> location, ValueShapeRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+  SortOp::Adaptor adaptor(operands, attributes, regions);
+  auto operandTypes = adaptor.operands().getTypes();
+  for (auto operandType : operandTypes) {
+    auto operandShapedType = operandType.cast<ShapedType>();
+    if (operandShapedType.hasRank()) {
+      int64_t cmpDim = adaptor.dimension();
+      int64_t rank = operandShapedType.getRank();
+      if (cmpDim < -rank || cmpDim >= rank)
```

Comment:
I am surprised that we're allowing negative axes here, because I think that everywhere else we require positive axes. I thought that this is accidental, but e.g. [the TF/XLA bridge relies on this](https://github.com/tensorflow/tensorflow/blob/f0bbe9a42cfd91cbf2551cdba7a1bab2d2cf5898/tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc#L5940). Well, one more thing to clean up when working on the dynamism RFC.

Reply:
Yes, and in ODS of SortOp, there is `CArg<"int64_t", "-1">:$dimension`. So this is intended. Sure, include this in dynamism RFC sounds good.